service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -297,11 +293,6 @@ INSERT INTO workspace (workspace_id, user_facing_id, display_name, description,
       }
     }
 
-    // If we have applicationIds to create, do that now.
-    if (applicationIds != null) {
-      applicationDao.enableWorkspaceApplications(workspaceUuid, applicationIds);
-    }
-
```

Comment:
this was the main problem for https://broadworkbench.atlassian.net/browse/WOR-1256. This is called in a function annotated with @WriteTransaction and then it calls a function in another dao that is also @WriteTransaction. The transaction propagation is REQUIRED so the same transaction is used but another retry interceptor is layered in! So we do retries within the same transaction instead of retrying the entire transaction.

I removed this code because it was causing problems and it is not right anyway. To enable apps we need to do the database work _and_ call sam. Since there is already a flight to do so, the solution is to change the create workspace flight to launch a subflight to enable apps if appropriate.